test/playlist-loader.test.js

```diff
@@ -469,1593 +423,2109 @@ QUnit.test('updateMaster changes old values', function(assert) {
-    'captured the responseText'
+
+  QUnit.test(
+    'does not misintrepret playlists missing newlines at the end',
+    function(assert) {
+      const loader = new PlaylistLoader('media.m3u8', this.fakeVhs);
+
+      loader.load();
+
+      // no newline
+      this.requests.shift().respond(
+        200, null,
+        '#EXTM3U\n' +
+        '#EXT-X-MEDIA-SEQUENCE:0\n' +
+        '#EXTINF:10,\n' +
+        'low-0.ts\n' +
+        '#EXT-X-ENDLIST'
+      );
+      assert.ok(loader.media().endList, 'flushed the final line of input');
+    }
   );
-});
 
-QUnit.test('switches media playlists when requested', function(assert) {
-  const loader = new PlaylistLoader('master.m3u8', this.fakeVhs);
+  QUnit.test('Supports multiple STREAM-INF with the same URI', function(assert) {
+    const loader = new PlaylistLoader('master.m3u8', this.fakeVhs);
 
-  loader.load();
+    loader.load();
 
-  this.requests.pop().respond(
-    200, null,
-    '#EXTM3U\n' +
-                              '#EXT-X-STREAM-INF:BANDWIDTH=1\n' +
-                              'low.m3u8\n' +
-                              '#EXT-X-STREAM-INF:BANDWIDTH=2\n' +
-                              'high.m3u8\n'
-  );
-  this.requests.pop().respond(
-    200, null,
-    '#EXTM3U\n' +
-                              '#EXT-X-MEDIA-SEQUENCE:0\n' +
-                              '#EXTINF:10,\n' +
-                              'low-0.ts\n'
-  );
+    this.requests.shift().respond(
+      200, null,
+      '#EXTM3U\n' +
+      '#EXT-X-STREAM-INF:BANDWIDTH=1,AUDIO="aud0"\n' +
+      'video/media.m3u8\n' +
+      '#EXT-X-STREAM-INF:BANDWIDTH=2,AUDIO="aud1"\n' +
+      'video/media.m3u8\n'
+    );
+    assert.equal(
+      loader.master.playlists['0-video/media.m3u8'].id,
+      loader.master.playlists[0].id,
+      'created key based on playlist id'
+    );
 
-  loader.media(loader.master.playlists[1]);
-  assert.strictEqual(loader.state, 'SWITCHING_MEDIA', 'updated the state');
+    assert.equal(
+      loader.master.playlists['1-video/media.m3u8'].id,
+      loader.master.playlists[1].id,
+      'created key based on playlist id'
+    );
+  });
 
-  this.requests.pop().respond(
-    200, null,
-    '#EXTM3U\n' +
-                              '#EXT-X-MEDIA-SEQUENCE:0\n' +
-                              '#EXTINF:10,\n' +
-                              'high-0.ts\n'
-  );
-  assert.strictEqual(loader.state, 'HAVE_METADATA', 'switched active media');
-  assert.strictEqual(
-    loader.media(),
-    loader.master.playlists[1],
-    'updated the active media'
-  );
-});
+  QUnit.module('llhls');
 
-QUnit.test(
-  'can switch playlists immediately after the master is downloaded',
-  function(assert) {
-    const loader = new PlaylistLoader('master.m3u8', this.fakeVhs);
+  QUnit.test('#EXT-X-SKIP does not add initial empty segments', function(assert) {
+    this.fakeVhs.options_ = {experimentalLLHLS: true};
+    const loader = new PlaylistLoader('media.m3u8', this.fakeVhs);
 
     loader.load();
```

Comment:
Might be worth pulling these all into a `beforeEach` since they're done in each test.